Documentation/grpc-metadata.md

```diff
@@ -223,3 +223,54 @@ func (s *server) SomeStreamingRPC(stream pb.Service_SomeStreamingRPCServer) erro
     stream.SetTrailer(trailer)
 }
 ```
+
+## Updating metadata in the interceptor - server side
+
+Server side metadata updating in the interceptor examples are available [here](../examples/features/metadata_interceptor/server/main.go).
+
+#### Unary interceptor
+
+To read and update metadata in the interceptor under unary call,
+the server needs to retrieve it from RPC context and update it through this RPC context.
+
+```go
+func SomeInterceptor(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+    // Get the incoming metadata from the RPC context, and add a new
+    // key-value pair to it.
+    md, ok := metadata.FromIncomingContext(ctx)
+    md.Append("key1", "value1")
+
+    // Create a context with the new metadata and pass it to handler.
+    ctx = metadata.NewIncomingContext(ctx, md)
+    return handler(ctx, req)
+}
+```
+
+#### Streaming interceptor
+
+There is no direct way to set RPC context in the stream interceptor.
```

Comment:
How about something like:

```
`grpc.ServerStream` does not provide a way to modify its RPC context. The streaming 
interceptor therefore needs to implement the `grpc.ServerStream` interface and return
a context with the updated metadata.

The easiest way to do this would be to create a type which embeds the `grpc.ServerStream`
interface and overrides only the `Context()` method to return a context with the updated
metadata. The streaming interceptor would then pass this wrapped stream to the provided handler.
```

Reply:
Got it, I will update the document